presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestSymbolMapper.java

```diff
@@ -0,0 +1,103 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.type.BigintType;
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder;
+import com.facebook.presto.sql.planner.plan.AggregationNode;
+import com.facebook.presto.sql.planner.plan.TopNNode;
+import com.facebook.presto.sql.planner.plan.ValuesNode;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.testing.Closeables.closeAllRuntimeException;
+import static java.util.Collections.singleton;
+import static java.util.Collections.singletonList;
+import static org.testng.Assert.assertEquals;
+
+public class TestSymbolMapper
+{
+    private static final Symbol a = new Symbol("a");
+    private static final Symbol b = new Symbol("b");
+    private static final Symbol c = new Symbol("c");
+    private static final Symbol A = new Symbol("A");
+    private static final Symbol B = new Symbol("B");
+
+    private static final SymbolMapper SYMBOL_MAPPER = SymbolMapper.builder()
+            .put(a, A)
+            .put(b, B)
+            .put(B, a)
+            .build();
+
+    private Metadata metadata;
+    private LocalQueryRunner queryRunner;
+
+    @BeforeClass
+    public final void setUp()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("tiny")
+                .setSystemProperty("task_concurrency", "1") // these tests don't handle exchanges from local parallel
+                .build();
+
+        queryRunner = new LocalQueryRunner(session);
+        this.metadata = queryRunner.getMetadata();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void tearDown()
+    {
+        closeAllRuntimeException(queryRunner);
+        queryRunner = null;
+        metadata = null;
+    }
+
+    @Test
+    public void testAggregationNodeSymbolMapping()
+    {
+        PlanBuilder builder = new PlanBuilder(new PlanNodeIdAllocator(), metadata);
+        ValuesNode values = builder.values(a, b);
+        AggregationNode aggregation = builder.aggregation(aggregationBuilder -> aggregationBuilder
+                .addGroupingSet(a, b)
+                .addAggregation(c, builder.expression("count(b)"), ImmutableList.of(BigintType.BIGINT))
+                .source(values));
```

Comment:
Shouldn't the aggregation node also have symbols for group id, hash, and an aggregation mask?